programs/ssl/ssl_server2.c

```diff
@@ -2593,21 +2618,46 @@ int main( int argc, char *argv[] )
                 goto exit;
             }
         }
-
-        if ( mbedtls_pk_get_type( &pkey2 ) == MBEDTLS_PK_ECKEY ||
-             mbedtls_pk_get_type( &pkey2 ) == MBEDTLS_PK_RSA )
+        else
         {
-            if( mbedtls_pk_get_type( &pkey2 ) == MBEDTLS_PK_ECKEY )
+            if ( mbedtls_pk_get_type( &pkey ) == MBEDTLS_PK_ECKEY ||
+                 mbedtls_pk_get_type( &pkey ) == MBEDTLS_PK_RSA )
```

Comment:
Instead of keeping the old code, can you set opt.key1_opaque_alg1 & opt.key1_opaque_alg2 accordingly if key_opaque_algs/key_opaque_algs2 wasn't passed ? And reuse key_opaque_set_alg_usage() for all cases

Reply:
Setting `opt.key1_opaque_alg1/opt.key1_opaque_alg2` would require to create new buffers, init them with strings corresponding to algs based on the key type and set `opt.key1_opaque_alg1/opt.key1_opaque_alg2` to point to the buffers. Then `key_opaque_set_alg_usage()` can be resused.
This also seems to not optimal solution.

Maybe it would be easier to extend `key_opaque_set_alg_usage()` and add `key_type*` parameter.
- if `alg1` and `alg2` are defined then use them to obtain psa `algs` and `usage`.
- otherwise use additional `key_type*` parameter to set default psa `algs` and `usage`.

Reply:
Either is OK, for my proposal setting the strings would be no cost since those strings are already defined in `key_opaque_alg_parse` and `key_opaque_set_alg_usage` and the linked will probably not duplicate them at the end.
It would be slighly less optimal because it would add some more `strcmp()`.

I fear the second solution would overcomplicate `key_opaque_set_alg_usage`.